src/main/java/com/varukha/onlinebookstore/service/book/impl/BookServiceImpl.java

```diff
@@ -59,10 +61,22 @@ public BookDto update(Long id, CreateBookRequestDto bookRequestDto) {
     @Override
     public List<BookDto> search(BookSearchParametersDto params) {
         Specification<Book> bookSpecification = bookSpecificationBuilder.build(params);
-        return bookRepository.findAll(bookSpecification)
-                .stream()
+        List<Book> books = bookRepository.findAll(
+                Specification.where(bookSpecification)
+                        .and((root, query, criteriaBuilder) -> {
```

Comment:
You have a specification builder from what I see. Shouldn't this logic with category criteria be there?